scheduler/recovery.go

```diff
@@ -2,21 +2,64 @@ package scheduler
 
 import (
 	"context"
+	"errors"
 	"reflect"
+	"time"
+
+	"github.com/puppetlabs/horsehead/v2/netutil"
+)
+
+const (
+	defaultBackoffMultiplier         = time.Millisecond * 5
+	defaultMaxRetries                = 10
```

Comment:
It might be good to have an infinite number of retries by default for backward compatibility. Maybe some sentinel value and a paired constant so people can opt in to that behavior manually if they want, too.

Reply:
Go tends to have negatives signal something like this. What if we made `MaxRetries < 0` mean infinite? Then `defaultMaxRetries = -1`.

see: https://golang.org/pkg/net/http/httputil/#ReverseProxy and https://golang.org/pkg/net/#Dialer for some examples of documented negatives changing behavior.

Reply:
I'd almost say that `<= 0` could mean infinite, since having 0 retries doesn't make sense - you'd just not use this descriptor at all in that case.